draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -717,6 +774,175 @@ Steps:
 23. Output (x, y)
 ~~~
 
+### Boneh-Franklin Method {#supersingular}
+
+The map2curve_bf(alpha) implements the Boneh-Franklin method {{BF01}} which
+covers the case of supersingular curves `E: y^2=x^3+B`.
+
+**Preconditions**
+
+This algorithm works for any Weierstrass curve over `F_q` such that `A=0` and
+`q=2 mod 3`.
+
+**Examples**
+
+- `y^2=x^3+1`
+
+**Algorithm**: map2curve_bf
+
+Input:
+
+ - `alpha`: an octet string to be hashed.
+ - `B`: the constant from the Weierstrass curve.
+
+Output:
+
+ - `(x,y)`: a point in E.
+
+Operations:
+
+~~~
+1. u = HashToBase(alpha)
+2. x = (u^2-B)^(1/3)
+3. Output (x,u)
```

Comment:
IIUC, the paper also describes multiplying by the cofactor (`l`). Do we need that? If not, why?

Reply:
Yes, it is required to multiply by a cofactor and this is recurrent in several encodings to obtain points in a subgroup of points. For the BF encoding, the value of the cofactor varies as is dependent on the target subgroup.